tests/integration_tests/modules/test_write_files.py

```diff
@@ -0,0 +1,71 @@
+"""Integration test for the write_files module.
+This test specifies files to be created by the ``write_files`` module
+and then checks if those files were created during boot.
+
+(This is ported from
+``tests/cloud_tests/testcases/modules/write_files.yaml``.)"""
+
+import pytest
+
+
+# NOTE: on trusty 'file' has an output formatting error for binary files and
+#       has 2 spaces in 'LSB  executable', which causes a failure here
+#
+# NOTE: the binary data can be any binary data, not only executables
+#       and can be generated via the base 64 command as such:
+#           $ base64 < hello > hello.txt
+#       the opposite is running:
+#           $ base64 -d < hello.txt > hello
+#
+USER_DATA = """\
+#cloud-config
+write_files:
+-   encoding: b64
+    content: CiMgVGhpcyBmaWxlIGNvbnRyb2xzIHRoZSBzdGF0ZSBvZiBTRUxpbnV4
```

Comment:
As we are defining this string in Python, what do you think to doing something like:

```py
ASCII_TEXT = "ASCII text"
B64_CONTENT = base64.b64encode(ASCII_TEXT)
```

and then using `.format()` to include it in `USER_DATA`?

This saves us from having the `"ASCII text"` string defined only in the assertion code, which I think makes it easier to parse what is going on.